### PR TITLE
sdk: add new hash pkg with ring member and hash implementation.

### PR DESCRIPTION
**Pull request by jrasell**

### CODE REVIEW

**Review by lgfa29 — APPROVED** [review]
The code LGTM, but I am curious if we actually need these interfaces since only the `ha` package would use them.

**Comment by jrasell** [comment]
> The code LGTM, but I am curious if we actually need these interfaces since only the `ha` package would use them.

I find they help keep the code structured and isolated even if they are simple and only to be used by a single caller. It also helps me personally design and write the code in a better style. We can revisit before merging to main if we find there are overriding factors why we shouldn't use these.

**Pull request merged**
